code/procs/scorestats.dm

```diff
@@ -352,6 +355,33 @@ var/datum/score_tracker/score_tracker
 
 		return jointext(., "")
 
+	//I feel like I may be reinventing the wheel here, but here's a recursive wheel
+	//recursively finds all items of type in container
+	proc/find_items_in(obj/container, type)
+		var/list/found = new /list(0)
+		for (var/obj/item/item in container.contents)
+			if (istype(item, type))
+				found += item
+			found += find_items_in(item, type)
+
+		return found
+
+	proc/get_inspector_report()
+		var/report = ""
+		//this might be a really stupid way of finding all players of a specific job, please tell me if it is
+		for (var/mob/living/carbon/human/player in mobs)
+			if (player.job != "Inspector")
+				continue
+			report += "<B>Inspector [player.real_name]'s report</B><BR><HR>"
+			//find all clipboards they have on them (probably just one, but may as well check)
+			var/list/clipboards = find_items_in(player, /obj/item/clipboard)
```

Comment:
The player is not an obj, but "find_items_in" typecasts that argument to an obj.

Reply:
Good point, is there a way I should type this to ensure that .contents exists? I'm not sure which base class it comes from.

Reply:
typecast it to `atom` and restrict it to obj/mob